src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -93,11 +97,22 @@ export class ElasticsearchService
     };
   }
   public async start(): Promise<InternalElasticsearchServiceStart> {
-    if (!this.client) {
+    if (!this.client || !this.esNodesCompatibility$) {
       throw new Error('ElasticsearchService needs to be setup before calling start');
     }
 
     const config = await this.config$.pipe(first()).toPromise();
+
+    // Log every error we may encounter in the connection to Elasticsearch
+    this.esNodesCompatibility$.subscribe(({ isCompatible, message }) => {
+      if (!isCompatible && message) {
+        this.log.error(message);
+      }
+    });
```

Comment:
It makes more sense to log these errors from the Elasticsearch service rather than the SavedObjects one. But also, we're now awaiting for a valid connection in this service before letting the SO service to start. So not having this logger here would lead to silent on-hold starts.